### DISCUSSION TITLE
Displaying image from local

### DISCUSSION THREAD

**Discussion by debjanipaul** [h2oai/wave#1467] [Q&A] [answered]
I am trying to add image stored locally to header_card and tall_info_card but the path I am providing is not working. I see the following error in console: 'Failed to load resource: the server responded with a status of 404 (Not Found)'. However if I use a path from internet it works. I have tried to provide absolute path but that didn't work too.

`q.page['page_content'] = ui.tall_info_card(
        box='page_content',
        title='Some text',
        image='./assets/landingPagepic.png',
        caption='Some text',
        name="content",
    )
`

**Comment by lo5** [comment, marked as answer]
Your question made me realize that this is not covered anywhere in the docs :)

Just added this section: https://github.com/h2oai/wave/blob/master/website/docs/files.md#serving-images

First `upload()` the image, then use the return value as the path.

**Reply by debjanipaul** [reply]
I tried the above mentioned method but getting the following error: 'raise ValueError(f'{name}: want one of {types}, got {type(value)}')
ValueError: TallInfoCard.image: want one of (<class 'str'>,), got <class 'list'>'

`image = await q.site.upload(['./assets/landingPagepic.png'])
        
    q.page['page_content'] = ui.tall_info_card(
        box='page_content',
        title='some text',
        image=image,
        caption='some text',
        name="page_content",
    )


`

**Reply by lo5** [reply]
You're missing a comma. `q.site.upload()` returns a tuple. See the example linked to above.
```py
image, = await q.site.upload(['path/to/my/image.png'])
```

**Reply by azim-b** [reply]
@debjanipaul, `await q.site.upload(['./assets/landingPagepic.png'])` returns a **list** of image paths on the server.

So what you want is **(** await q.site.upload(['./assets/landingPagepic.png']) **)[0]** to get the first (and only) returned path.

Here's a link to the documentation: https://wave.h2o.ai/docs/files#serving-images

**Reply by debjanipaul** [reply]
Thanks @azim-b. It worked.
